pkg/controller/che/tls-secrets.go

```diff
@@ -0,0 +1,287 @@
+	const cheTLSSelfSignedCertificateSecretName string = "self-signed-certificate"
+
+	// ===== Check Che TLS certificate ===== //
+
+	cheTLSSecret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: cheTLSSecretName}, cheTLSSecret)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			// Error reading secret info
+			logrus.Errorf("Error getting Che secert \"%s\": %v", cheTLSSecretName, err)
+			return reconcile.Result{}, err
+		}
+
+		// Che TLS secret doesn't exist, generate a new one
+
+		// Remove Che CA certificate secret if any
+		cheCASelfSignedCertificateSecret := &corev1.Secret{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: cheTLSSelfSignedCertificateSecretName}, cheCASelfSignedCertificateSecret)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				// Error reading secret info
+				logrus.Errorf("Error getting Che secert \"%s\": %v", cheTLSSecretName, err)
+				return reconcile.Result{}, err
+			}
+			// Che CA certificate doesn't exists (that's expected at this point), do nothing
+		} else {
+			// Remove Che CA secret because Che TLS secret is missing (they should be generated together).
+			if err = r.client.Delete(context.TODO(), cheCASelfSignedCertificateSecret); err != nil {
+				logrus.Errorf("Error deleting Che TLS secret: %v", err)
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Prepare permissions for the certificate generation job
+
+		const cheTLSJobServiceAccountName string = "che-tls-job-service-account"
+		cheTLSJobServiceAccount := &corev1.ServiceAccount{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: cheTLSJobServiceAccountName}, cheTLSJobServiceAccount)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				logrus.Errorf("Error getting Che TLS job service account \"%s\": %v", cheTLSJobServiceAccountName, err)
+				return reconcile.Result{}, err
+			}
+
+			// Che TLS job service accound doesn't exist, creating one
+			cheTLSJobServiceAccount = deploy.NewServiceAccount(checluster, cheTLSJobServiceAccountName)
+			if err := r.CreateServiceAccount(checluster, cheTLSJobServiceAccount); err != nil {
+				logrus.Errorf("Error creating Che TLS job service account \"%s\": %v", cheTLSJobServiceAccountName, err)
+				return reconcile.Result{}, err
+			}
+		}
+
+		const cheTLSJobRoleName string = "che-tls-job-role"
+		cheTLSJobRole := &rbac.Role{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: cheTLSJobRoleName}, cheTLSJobRole)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				logrus.Errorf("Error getting Che TLS job role \"%s\": %v", cheTLSJobRoleName, err)
+				return reconcile.Result{}, err
+			}
+
+			// Che TLS job role doesn't exist, creating one
+			cheTLSJobRole = deploy.NewRole(checluster, cheTLSJobRoleName, []string{"secrets"}, []string{"create"})
```

Comment:
Could we apply this permission here https://github.com/eclipse/che-operator/blob/master/deploy/role.yaml ? Then we don't need to grammatically create it...

Reply:
Already done [here](https://github.com/eclipse/che-operator/pull/220/files#diff-4374306bd7fd15cc09428b7ccf9bbec8R23). But I am not sure that child objects can reuse parent permissions, am I wrong?

Reply:
Ok minimal permissions per job make sense. About question: You could create role binding for this job with secrets only because parent already has this permissions. Parent can't provide more permission to resource object than it has itself.

Reply:
Thank you for answering, but my question was a bit different: whether child object has the same permissions as its parent. I don't think so, but not sure.